### DISCUSSION TITLE
Jinja template substitution not working in DataplexCreateOrUpdateDataQualityScanOperator

### DISCUSSION THREAD

**Discussion by sukalpv** [apache/airflow#36995] [Q&A]
The [documentation](https://airflow.apache.org/docs/apache-airflow-providers-google/stable/_api/airflow/providers/google/cloud/operators/dataplex/index.html#airflow.providers.google.cloud.operators.dataplex.DataplexCreateOrUpdateDataProfileScanOperator) says `body` supports jinja templates. However, var `{{ ds }}` is not getting substituted at run time. Here is the snippet:
```
data_quality_scan.data_quality_spec = DataQualitySpec({
        'rules': [
            {
                'table_condition_expectation': {
                    'sql_expression': f'''MAX(event_ts) > {{ ds }} 23:00:00 UTC '''
                },
                'dimension': 'VALIDITY',
            },
        ],
    })
```
provider version is `10.13.0`.

**Comment by Sirivasv** [comment]
Have you tried with `{ ds }` with single brackets, guessing as perhaps is using a task with the limitations described here:
https://airflow.apache.org/docs/apache-airflow/stable/templates-ref.html#accessing-airflow-context-variables-from-taskflow-tasks

**Comment by andresitorusz** [comment]
You need to add extra brackets if the jinja template is in f-string. Try this:

```
data_quality_scan.data_quality_spec = DataQualitySpec({
        'rules': [
            {
                'table_condition_expectation': {
                    'sql_expression': f'''MAX(event_ts) > {{{{ ds }}}} 23:00:00 UTC '''
                },
                'dimension': 'VALIDITY',
            },
        ],
    })
```

**Reply by sukalpv** [reply]
Didn't worked. I tried it. already

**Reply by andresitorusz** [reply]
Have you tried to format the value outside the DataQualitySpec? I mean like this:

```
sql_expression = f'''MAX(event_ts) > {{{{ ds }}}} 23:00:00 UTC '''

data_quality_scan.data_quality_spec = DataQualitySpec({
        'rules': [
            {
                'table_condition_expectation': {
                    'sql_expression': sql_expression 
                },
                'dimension': 'VALIDITY',
            },
        ],
    })
```

or like this:

```
sql_expression = f'MAX(event_ts) > {{ ds }} 23:00:00 UTC '

data_quality_scan.data_quality_spec = DataQualitySpec({
        'rules': [
            {
                'table_condition_expectation': {
                    'sql_expression': sql_expression 
                },
                'dimension': 'VALIDITY',
            },
        ],
    })
```

**Reply by sukalpv** [reply]
didn't work unfortunately

**Reply by andresitorusz** [reply]
Can you provide the screenshot of the task rendered template? And your Airflow version maybe.

Here is how you can check the rendered template:

![image](https://github.com/apache/airflow/assets/36465150/930e35ff-cdb7-4048-ba88-a7402c5160d5)

**Reply by sukalpv** [reply]
![image](https://github.com/apache/airflow/assets/101869570/320c05e6-9373-4042-8169-4aadf9c2ad9f)
With f-strings template changes to `{{ ds }}`, but not getting substituted
